### PR TITLE
Ug/strategy pattern

### PR DESCRIPTION
**Pull request by mizge**

### CODE REVIEW

File: src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using TowerDefense.Api.Models;
+using TowerDefense.Api.Models.Player;
+
+namespace TowerDefense.Api.Strategies.RocketStrategy
+{
+    public class SecondLevelRocketStrategy : IRocketsStrategy
+    {
+        public List<AttackResult> Attack(GridItem[] opponentGridItems, IPlayer opponent, int damage, int rocketGridId)
+        {
+            int affectedRow = rocketGridId / 8;
```
Comment (antanasramana) [inline src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs:11]: what does this do? Attack in a line, attack in a curve. Please use method which defines this strategy purpose. :)

Comment (antanasramana) [reply src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs:11]: And update other strategies as well

File: src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            List<int> affectedGridItems = Enumerable.Range(0, 8).Select(x => x + (affectedRow + 1 * 8)).Reverse().ToList();
+            affectedGridItems.AddRange(Enumerable.Range(0, 8).Select(x => x + (affectedRow - 1 * 8)).Reverse().ToList());
+            GridItem[] opponentsAffectedGridItems = opponentGridItems.Where(x => affectedGridItems.Contains(x.Id))
+                                                                     .OrderByDescending(x => x.Id)
+                                                                     .ToArray();
+            List<AttackResult> result = new List<AttackResult>();
```
Comment (antanasramana) [inline src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs:17]: please use vars where possible. Most of the solution is written in that way so please be consistent

File: src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs (hunk 1 above)
Comment (antanasramana) [inline src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs:11]: magic numbers everywhere - please use constants to define their purpose

File: src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs (hunk 2 above)
Comment (antanasramana) [reply src/TowerDefense.Api/Strategies/RocketStrategy/SecondLevelRocketStrategy.cs:17]: we have game constants class. Which contains I think your value that you have used. Take a look there

File: src/TowerDefense.Api/Models/Items/IItem.cs [hunk 3]
```diff
@@ -4,6 +4,9 @@ public interface IItem
     {
         string Id { get; set; }
         int Price { get; set; }
+        int Health { get; set; }
+        int Demage { get; set; }
```
Comment (antanasramana) [inline src/TowerDefense.Api/Models/Items/IItem.cs:8]: Typo

File: src/TowerDefense.Api/Strategies/RocketStrategy/ThirdLevelRocketStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using TowerDefense.Api.Battle.Grid;
+using TowerDefense.Api.Models;
+using TowerDefense.Api.Models.Player;
+
+namespace TowerDefense.Api.Strategies.RocketStrategy
+{
+    public class ThirdLevelRocketStrategy : IRocketsStrategy
```
Comment (antanasramana) [inline src/TowerDefense.Api/Strategies/RocketStrategy/ThirdLevelRocketStrategy.cs:7]: It's being toggled by the level not by what the strategy does

**Review by antanasramana — CHANGES_REQUESTED** [review]

**Review by tomvai15 — APPROVED** [review]

**Review by antanasramana — APPROVED** [review]

**Pull request merged**